ledger/src/slot_stats.rs

```diff
@@ -0,0 +1,90 @@
+use {
+    crate::blockstore_meta::SlotMeta, bitflags::bitflags, lru::LruCache, solana_sdk::clock::Slot,
+};
+
+const SLOTS_STATS_CACHE_CAPACITY: usize = 300;
+
+macro_rules! get_mut_entry (
+    ($cache:expr, $key:expr) => (
+        match $cache.get_mut(&$key) {
+            Some(entry) => entry,
+            None => {
+                $cache.put($key, SlotStats::default());
+                $cache.get_mut(&$key).unwrap()
+            }
+        }
+    );
+);
+
+#[derive(Copy, Clone, Debug)]
+pub(crate) enum ShredSource {
+    Turbine,
+    Repaired,
+    Recovered,
+}
+
+bitflags! {
+    #[derive(Default)]
+    struct SlotFlags: u8 {
```

Comment:
nit: rename to `SlotState`

Reply:
There are already `SlotStats` and `SlotsStats` in this file.
So I guess `SlotState` would be bit confusing and easy to misread here.

Also we are using "flags" elsewhere in the code; e.g.
https://github.com/solana-labs/solana/blob/91c272985/sdk/src/packet.rs#L17-L36
https://github.com/solana-labs/solana/blob/91c272985/ledger/src/shred.rs#L215

Reply:
what about `SlotStats -> SlotMetrics` and then `SlotFlags -> SlotState`